x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.test.ts

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { LogicMounter } from '../../../__mocks__/kea.mock';
+
+jest.mock('../../../shared/kibana', () => ({
+  KibanaLogic: { values: { history: { location: { search: '' } } } },
+}));
+import { KibanaLogic } from '../../../shared/kibana';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+
+jest.mock('../../../shared/flash_messages', () => ({
+  flashAPIErrors: jest.fn(),
+}));
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+jest.mock('../engine', () => ({
+  EngineLogic: { values: { engineName: 'test-engine' } },
+}));
+
+import { AnalyticsLogic } from './';
+
+describe('AnalyticsLogic', () => {
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    analyticsUnavailable: false,
+  };
+
+  const MOCK_TOP_QUERIES = [
+    {
+      doc_count: 5,
+      key: 'some-key',
+    },
+    {
+      doc_count: 0,
+      key: 'another-key',
+    },
+  ];
+  const MOCK_RECENT_QUERIES = [
+    {
+      document_ids: ['1', '2'],
+      query_string: 'some-query',
+      tags: ['some-tag'],
+      timestamp: 'some-timestamp',
+    },
+  ];
+  const MOCK_TOP_CLICKS = [
+    {
+      key: 'highly-clicked-query',
+      doc_count: 1,
+      document: {
+        id: 'some-id',
+        engine: 'some-engine',
+        tags: [],
+      },
+      clicks: {
+        doc_count: 100,
+      },
+    },
+  ];
+  const MOCK_ANALYTICS_RESPONSE = {
+    analyticsUnavailable: false,
+    allTags: ['some-tag'],
+    recentQueries: MOCK_RECENT_QUERIES,
+    topQueries: MOCK_TOP_QUERIES,
+    topQueriesNoResults: MOCK_TOP_QUERIES,
+    topQueriesNoClicks: MOCK_TOP_QUERIES,
+    topQueriesWithClicks: MOCK_TOP_QUERIES,
+    totalClicks: 1000,
+    totalQueries: 5000,
+    totalQueriesNoResults: 500,
+    clicksPerDay: [0, 10, 50],
+    queriesPerDay: [10, 50, 100],
+    queriesNoResultsPerDay: [1, 2, 3],
+  };
+  const MOCK_QUERY_RESPONSE = {
+    analyticsUnavailable: false,
+    allTags: ['some-tag'],
+    totalQueriesForQuery: 50,
+    queriesPerDayForQuery: [25, 0, 25],
+    topClicksForQuery: MOCK_TOP_CLICKS,
+  };
+
+  const { mount } = new LogicMounter(AnalyticsLogic);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    KibanaLogic.values.history.location.search = '';
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(AnalyticsLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('onAnalyticsUnavailable', () => {
+      it('should set state', () => {
```

Comment:
@JasonStoltz - I implemented the reducer block suggestions you noted in https://github.com/elastic/kibana/pull/87561#discussion_r554010592! I like this a lot so far, feels much faster 🎉